core/src/test/java/io/lionweb/lioncore/java/serialization/MyNodeWithStructuredDataType.java

```diff
@@ -0,0 +1,65 @@
+package io.lionweb.lioncore.java.serialization;
+
+import io.lionweb.lioncore.java.language.*;
+import io.lionweb.lioncore.java.model.ClassifierInstanceUtils;
+import io.lionweb.lioncore.java.model.StructuredDataTypeInstance;
+import io.lionweb.lioncore.java.model.impl.DynamicNode;
+
+public class MyNodeWithStructuredDataType extends DynamicNode {
```

Comment:
Maybe add an example (and tests) for nested SDTs, and SDTs containing enums.

In C#, I use the following example:

```
enum Currency
  EUR
  GBP

sdt Decimal
  int: Integer
  frac: Integer

sdt Amount
  value: Decimal
  currency: Currency
  digital: Boolean
```

Reply:
I added such example and two tests covering it